### PR TITLE
Fix Security Vulnerabilities

### PR DESCRIPTION
**Pull request by jocelynsaysrawr**
- Fixed security vulnerabilities.
- Migrated lint task in gulpfile from jscs to eslint

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,19 +17,17 @@
...
-    "gulp": "^3.8.8",
-    "gulp-jscs": "^1.1.2",
-    "gulp-jshint": "^1.8.4",
-    "gulp-load-plugins": "^0.6.0",
-    "gulp-mocha": "^1.1.0",
+    "coveralls": "^3.0.2",
+    "gulp": "^4.0.0",
+    "gulp-load-plugins": "^1.5.0",
+    "gulp-mocha": "^6.0.0",
     "gulp-plumber": "^0.6.5",
```
Comment (taesup) [inline package.json]: Do we still need coveralls, plumber, and sourcemaps?

**Comment by taesup** [comment]
Also, it's probably best in the future to just pull down this repo, make a branch from this repo and PR back to this repo so that someone else can pull the code down and try to run stuff.

File: package.json (hunk 1 above)
Comment (jocelynsaysrawr) [reply package.json]: I don't think so. I'm not really sure what those are used for.

File: gulpfile.js [hunk 2]
```diff
@@ -17,25 +17,32 @@ if (process.env.CI) {
...
+  console.log(err.toString());
+  this.emit('end');
+}
+
 gulp.task('lint', function(done) {
   return gulp.src(paths.lint)
     .pipe(plugins.eslint())
     .pipe(plugins.eslint.format())
-    .pipe(plugins.eslint.failAfterError());
+    // .pipe(plugins.eslint.failAfterError());
```
Comment (jaywon) [inline gulpfile.js]: Should just remove if not needed. 🔥

File: package.json [hunk 3]
```diff
@@ -17,17 +17,14 @@
...
   "dependencies": {
-    "debug": "^4.1.0",
-    "gulp-eslint": "^5.0.0"
+    "debug": "^4.1.0"
   },
   "devDependencies": {
     "chai": "^2.3.0",
-    "coveralls": "^3.0.2",
     "gulp": "^4.0.0",
+    "gulp-eslint": "^5.0.0",
```
Comment (jaywon) [inline package.json:25]: Nice catch on dependencies 👍

**Review by jaywon — APPROVED** [review]

**Review by taesup — APPROVED** [review]

**Pull request merged**